scripts/assert-changed-files.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+GREP_PATTERN=$1
+
+CHANGED_FILES="$(git diff master... --name-only -r | grep -E "$GREP_PATTERN")"
```

Comment:
This was _kind of_ exhibiting weird behavior on CircleCI. This looks accurate to more Unix-y 👀 than my own right?

Only thing I wasn't sure of is whether I'd need to fetch origin for this to work.